src/Service/Resolvers/Sql Query Structures/SqlUpsertQueryStructure.cs

```diff
@@ -88,6 +88,15 @@ public class SqlUpsertQueryStructure : BaseSqlQueryStructure
             // Populates the UpsertQueryStructure with UPDATE and INSERT column:value metadata
             PopulateColumns(mutationParams, sourceDefinition, isIncrementalUpdate: incrementalUpdate);
 
+            if (FieldsReferencedInDbPolicyForCreateAction.Count > 0)
+            {
+                // This indicates that one or more fields referenced in the database policy are not a part of the insert statement.
```

Comment:
An earlier PR of your #1391 had a good description for why this check is necessary and would be a good comment to have here and/or on the baseSqlQuerystructure where FieldsReferencedInDbPolicyForCreateAction is defined:
>will store all the fields
referenced in the db policy for create action. As we add columns to the
`InsertColumns` list for `SqlInsertQueryStructure` (PUT/PATCH/POST
operations) one by one, we remove the columns if present from the
`FieldsReferencedInDbPolicyForCreateAction`. So, at the end if the size
of this set `FieldsReferencedInDbPolicyForCreateAction` is 0, it means
that all the fields referenced in the database policy for create action
are being included in the insert statement, and we are good. However, if
the size is non-zero, we throw a bad request exception.